src/test/java/org/opensearch/geospatial/ip2geo/Ip2GeoTestCase.java

```diff
@@ -149,6 +161,34 @@ public File sampleIp2GeoFile() {
         return new File(this.getClass().getClassLoader().getResource("ip2geo/sample_valid.csv").getFile());
     }
 
+    public Datasource randomDatasource() {
+        Datasource datasource = new Datasource();
+        datasource.setName(GeospatialTestHelper.randomLowerCaseString());
+        datasource.setSchedule(
+            new IntervalSchedule(Instant.now().truncatedTo(ChronoUnit.MILLIS), Randomness.get().nextInt(30) + 1, ChronoUnit.DAYS)
+        );
+        datasource.setState(randomState());
+        datasource.setIndices(Arrays.asList(GeospatialTestHelper.randomLowerCaseString(), GeospatialTestHelper.randomLowerCaseString()));
+        datasource.setEndpoint(GeospatialTestHelper.randomLowerCaseString());
+        datasource.getDatabase()
+            .setFields(Arrays.asList(GeospatialTestHelper.randomLowerCaseString(), GeospatialTestHelper.randomLowerCaseString()));
+        datasource.getDatabase().setProvider(GeospatialTestHelper.randomLowerCaseString());
+        datasource.getDatabase().setUpdatedAt(Instant.now().truncatedTo(ChronoUnit.MILLIS));
+        datasource.getDatabase().setMd5Hash(GeospatialTestHelper.randomLowerCaseString());
+        datasource.getDatabase().setValidForInDays(Randomness.get().nextInt(30) + 1l);
+        datasource.getUpdateStats().setLastSkippedAt(Instant.now().truncatedTo(ChronoUnit.MILLIS));
+        datasource.getUpdateStats().setLastSucceededAt(Instant.now().truncatedTo(ChronoUnit.MILLIS));
+        datasource.getUpdateStats().setLastFailedAt(Instant.now().truncatedTo(ChronoUnit.MILLIS));
```

Comment:
I see many `Instant.now()` in code, can we create a local variable to hold the current Instant value, and use that variable everywhere it is needed?